content/react.md

```diff
@@ -0,0 +1,271 @@
+
+<h4 id="passing-callbacks-from-blaze">Passing callbacks to a React component</h4>
+
+To pass a callback to a React component that you are including with this helper, simply make a [template helper that returns a function](http://guide.meteor.com/blaze.html#pass-callbacks), and pass it in as a prop, like so:
+
+```js
+Template.userDisplay.helpers({
+  onClick() {
+    var self = Template.instance();
+
+    // Return a function from this helper, where the template instance is in
+    // a closure
+    return function () {
+      self.hasBeenClicked.set(true)
+    }
+  }
+});
+```
+
+To use it in Blaze:
+
+```js
+<template name="userDisplay">
+  <div>{{> React component=UserAvatar userId=_id onClick=onClick}}</div>
+</template>
+```
+
+<h2 id="livedata">Using data</h2>
+
+React is a front-end rendering library and as such doesn't concern itself with how data gets into and out of the component heirarchy. Meteor has strong opinions about data of course! Meteor's Livedata system defines a system [publications](data-loading.html) and [methods](methods.html) to subscribe to and modify the data in your application. 
+
+To combine the two systems, we've developed a [`react-meteor-data`](https://atmospherejs.com/meteor/react-meteor-data) package which allows React components to respond to data changes via Meteor's [Tracker](https://www.meteor.com/tracker) reactivity system.
+
+<h3 id="using-createContainer">Using `createContainer`</h3>
+
+Once you've run `meteor add react-meteor-data`, you'll be able to import the `createContainer` function, which allows you to create a [container component](https://medium.com/@dan_abramov/smart-and-dumb-components-7ca2f9a7c7d0#.by86emv9b) which provides data to your presentational components.
+
+(Note that "container components" are analogous to the "smart components" and "presentational components" to the "reusable components" in the pattern we document in the [UI/UX article](http://guide.meteor.com/ui-ux.html#components), if you'd like to read more about how this philosophy marries with Meteor).
+
+For example, in the Todos example app, we have a `ListPage` component, which renders the metadata about a list alongside the todos that are within it. In order to do so, it needs to [subscribe](data-loading.html#subscriptions) to the `todos.inList` publication, check that subscription's readiness, then fetch the list of todos from the `Todos` collection.
+
+It also needs to be responsive to reactive changes in the state of those actions (for instance if a todo changes due to the action of another user). All this data loading complexity is a typical use-case for a container-presentational component split, and the `createContainer()` function makes it simple to do this.
+
+We simply define the `ListPage` component to be a presentational component that expects it's data to be passed in as a property:
```

Comment:
IDK - I feel like the Guide should be correct and it's fine if the example app is slightly wrong?